target/riscv/cpu_helper.c

```diff
@@ -113,7 +114,61 @@ bool riscv_cpu_fp_enabled(CPURISCVState *env)
     return false;
 }
 
-void riscv_cpu_swap_hypervisor_regs(CPURISCVState *env)
+#ifdef TARGET_CHERI
+static void riscv_cpu_swap_hypervisor_capregs(
+    CPURISCVState *env, bool current_virt)
+{
+    if (current_virt) {
+        /* Current V=1 and we are about to change to V=0 */
```

Comment:
Could this be de-deduplicated using macros like the COPY_SPECIAL? I'm also not sure factoring this out is a good idea since we won't get merge conflicts when the hypervisor stuff changes. I assume it will change a bit in the future since it's still very new.

Reply:
I can de-duplicate it using COPY_SPECIAL approach, I opted to do it this way to avoid having COPY_SPECIAL with tons of parameters, since I would have had to add the CSR_* number and the CheriSCR_* number as well. I will revert back to inlining these as getting merge conflicts for hypervisor changes will definitely help during merges.